src/Socket.php

```diff
@@ -474,7 +474,11 @@ public function getType()
      */
     public function assertAlive()
     {
-        $code = $this->getOption(SOL_SOCKET, SO_ERROR);
+        if (is_resource($this->resource)) {
+            $code = $this->getOption(SOL_SOCKET, SO_ERROR);
+        } else {
+            $code = false;
+        }
```

Comment:
This will only solve this specific occurrence, but the broader problem still exists: After calling `close()`, any method call on this class could work on a closed resource and should throw an `Exception` as documented. It's my understanding this should be solved in a more generic way, i.e. all methods already have a check in place to report an `Exception` if the socket function call fails, but we might pass an invalid resource to `Exception::createFromSocketResource()`.